src/som/interpreter/actors/Actor.java

```diff
@@ -182,11 +187,34 @@ protected void appendToMailbox(final EventualMessage msg) {
     mailboxExtension.append(msg);
   }
 
-  protected static void handleBreakPoints(final EventualMessage msg, final WebDebugger dbg) {
-    if (VmSettings.TRUFFLE_DEBUGGER_ENABLED && msg.isBreakpoint()) {
-      dbg.prepareSteppingUntilNextRootNode();
+  protected static void handleBreakpointsAndStepping(final EventualMessage msg, final WebDebugger dbg) {
+    if (VmSettings.TRUFFLE_DEBUGGER_ENABLED) {
+      // check step-into and breakpoint before the message is processed
+      if (msg.isStepInto()) {
```

Comment:
if you feel that the comment is needed here, perhaps the method `isStepInto` can be renamed so that the comment isn't necessary anymore.